### PR TITLE
Refactor dist tests

### PR DESCRIPTION
**Pull request by farodin91**
* Split up AbstractJanusGraphAssemblyIT into a generic class for all release tests and suite for default backend tests
* activate single vertex tests
* centralized naming in pom.xml for zip names

Signed-off-by: Jan Jansen <jan.jansen@gdata.de>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java [hunk 1]
```diff
@@ -44,115 +26,98 @@
...
-        final boolean debug = false;
-        ImmutableMap<String, String> contextVars = ImmutableMap.of("janusgraphServerConfig", janusgraphServerConfig);
-        unzipAndRunExpect("janusgraph-server-sh.before.expect.vm", contextVars, full, debug);
-        assertTimeout(Duration.ofSeconds(30), () -> {
-            while (!serverListening("localhost",8182)) {
-                Thread.sleep(1000);
-            }
-        });
+    @Test
+    @Tag(TestCategory.FULL_TESTS)
```
Comment (li-boxuan) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:62]: Question: what is the purpose of this tag and how does it work?

**Review by li-boxuan — COMMENTED** [review]
Hi, @farodin91 Can you make the commit message a bit more elaborative, or leave a comment explaining what this PR aims to achieve? That would help with the review process.

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java (hunk 1 above)
Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:62]: This tag is just a preparation for the java 11 to filter out full tests.

**Comment by farodin91** [comment]
@li-boxuan Tried to update the commit message.

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/TestCategory.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.pkgtest;
+
+public class TestCategory {
```
Comment (li-boxuan) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/TestCategory.java:17]: There is also a TestCategory in janusgraph-backend-testutils module.

Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/TestCategory.java:17]: It is container for constants. I could rename it to PkgTestCategory

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java [hunk 3]
```diff
@@ -44,111 +26,109 @@
...
-        ZIPFILE_EXTRACTED = ZIPFILE_PATH.substring(0, ZIPFILE_PATH.length() - 4);
-        ZIPFILE_FULL_PATH = props.getProperty("zipfile-full.path");
-        ZIPFILE_FULL_EXTRACTED = ZIPFILE_FULL_PATH.substring(0, ZIPFILE_FULL_PATH.length() - 4);
+    protected abstract String getGraphName();
 
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
+    protected Boolean supportsIndex() {
+        return false;
```
Comment (li-boxuan) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:53]: I am just thinking... shall we make it as true by default? If we make it true, then if someone adds a subclass in the future which **does not** support index, the test case will immediately fail and they would know that they should override this method. On contrary, let's say we make it false by default, then if someone adds a subclass in the future which **does** support index (without noting they should override this method), index tests will be skipped, which is not ideal.

File: janusgraph-dist/pom.xml [hunk 4]
```diff
@@ -684,29 +690,22 @@
...
                                     <!--
                                         We set attached=true so that the gpg plugin will
-                                        sign all of our assembly artifacts. We handle
-                                        deployment of the artifacts manually because
-                                        maven-deploy-plugin can't upload files to a flat
-                                        S3 directory; it insists on creating a
-                                        long/packagename/and/artifactid/and/version/
-                                        path at the destination before uploading, and we
-                                        don't want that for our files S3. codehaus's
-                                        wagon:upload plugin comes to our rescue.
```
Comment (li-boxuan) [inline janusgraph-dist/pom.xml:701]: I am not familiar with this part. Can you explain why you deleted this paragraph?

**Review by li-boxuan — COMMENTED** [review]
In your commit message you mentioned, "activate single vertex tests". Could you point out what part does this? I think I missed it.

File: janusgraph-dist/pom.xml (hunk 4 above)
Comment (farodin91) [reply janusgraph-dist/pom.xml:701]: I just saw this comment which make no sense. We are not using s3 save some files.

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java (hunk 3 above)
Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:53]: i will check

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java [hunk 5]
```diff
@@ -44,111 +26,109 @@
...
-        }
-        finally
-        {
-            if(s != null)
-                try {s.close();}
-                catch(Exception e){}
+    @Test
+    @Tag(PkgTestCategory.FULL_TESTS)
+    public void testSimpleGremlinSessionFull() throws Exception {
+        unzipAndRunExpect("single-vertex.expect.vm", getConfigPath(), getGraphName(), true, false);
```
Comment (farodin91) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:64]: @li-boxuan This test wasn't activated before.

Comment (li-boxuan) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:64]: I can see "testSimpleGremlinSessionFull" in the old tests. Am I missing something?

Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:64]: I will rework the naming
I'm pretty sure the file single-vertex.expect was executed.

**Comment by farodin91** [comment]
@li-boxuan I debug using the github actions.

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java [hunk 6]
```diff
@@ -44,111 +26,90 @@
...
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
-    }
+    protected abstract String getGraphName();
 
-    protected void testSimpleGremlinSession(String graphConfig, String graphToString, boolean full) throws Exception {
-        unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);
+    @ParameterizedTest
+    @ArgumentsSource(EnvironmentProvider.class)
```
Comment (li-boxuan) [inline janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:57]: Should this be tagged with "FULL_TESTS"?

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java (hunk 5 above)
Comment (li-boxuan) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java]: > I'm pretty sure the file single-vertex.expect was executed.

You mean "was executed" or "was not executed"? In the old tests, I see a few invocations of method `testSimpleGremlinSession` which calls `unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);` My confusion is what exactly was missing before.

File: janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java (hunk 6 above)
Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:57]: I don't think so.

The tag full_tests should be just used in the provider to just run full tests under the condition that the tag full exists.

Comment (li-boxuan) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:57]: If I am understanding correctly, according to the logic in EnvironmentProvider, the parameter `fullTest` will be always false unless it is tagged with `FULL_TESTS`. 

In the old code, however, I can see `single-vertex.expect.vm` is tested with `fullTest` being true & false.
```
    @Test
    public void testSimpleGremlinSession() throws Exception {
        testSimpleGremlinSession("conf/janusgraph-berkeleyje.properties", "berkeleyje", false);
    }

    @Test
    public void testSimpleGremlinSessionFull() throws Exception {
        testSimpleGremlinSession("conf/janusgraph-berkeleyje.properties", "berkeleyje", true);
    }
```

Thus, it seems to me that the new test is only covering the case when `fullTest == false`, while the old test covered both cases. Is this by intention?

Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:57]: I add the handle to able to disable the full tests. It should execute same tests currently. I'll check it.

Comment (farodin91) [reply janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java:57]: I'll remove this logic with selecting tests and add it if need.

**Comment by farodin91** [comment]
@li-boxuan I switched away from parametrized testing solution. Now, I';m able to filter all full tests by searching for `*Full`. We did something similar for integration tests.

**Comment by li-boxuan** [comment]
> Now, I'm able to filter all full tests by searching for *Full. We did something similar for integration tests.

I don't quite understand how that helps/works. Can you elaborate?

**Comment by farodin91** [comment]
> I don't quite understand how that helps/works. Can you elaborate?

I can now use exclude tests in surefire to exclude all test that are running against full zip instead of small step.

**Review by li-boxuan — APPROVED** [review]
LGTM 👍

**Pull request merged**